programs/drift/src/instructions/admin.rs

```diff
@@ -1663,6 +1664,82 @@ pub fn handle_deposit_into_perp_market_fee_pool<'c: 'info, 'info>(
     Ok(())
 }
 
+#[access_control(
+    spot_market_valid(&ctx.accounts.spot_market)
+)]
+pub fn handle_deposit_into_spot_market_vault<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, DepositIntoSpotMarketVault<'info>>,
+    amount: u64,
+) -> Result<()> {
+    let spot_market = &mut load_mut!(ctx.accounts.spot_market)?;
+
+    let remaining_accounts_iter = &mut ctx.remaining_accounts.iter().peekable();
+
+    let mint = get_token_mint(remaining_accounts_iter)?;
+
+    msg!(
+        "depositing {} into spot market {} vault",
+        amount,
+        spot_market.market_index
+    );
+
+    let deposit_token_amount_before = spot_market.get_deposits()?;
+
+    let deposit_token_amount_after = deposit_token_amount_before.safe_add(amount.cast()?)?;
+
+    validate!(
+        deposit_token_amount_after > deposit_token_amount_before,
+        ErrorCode::DefaultError,
+        "new_deposit_token_amount ({}) <= deposit_token_amount ({})",
+        deposit_token_amount_after,
+        deposit_token_amount_before
+    )?;
+
+    let token_precision = spot_market.get_precision();
+
+    let cumulative_deposit_interest_before = spot_market.cumulative_deposit_interest;
+
+    let cumulative_deposit_interest_after = deposit_token_amount_after
+        .safe_mul(SPOT_CUMULATIVE_INTEREST_PRECISION)?
+        .safe_div(spot_market.deposit_balance)?
+        .safe_mul(SPOT_BALANCE_PRECISION)?
+        .safe_div(token_precision.cast()?)?;
```

Comment:
maybe good to enforce things to avoid large int changes like:
1) maximum deposit amounts (based on current `cumulative_deposit_interest`)

also check if max deposits is violated